File: 240208/G or H 2/G-or-H-2.py
n=int(input())

arr=[-1]*101
cnt=[0,0]

for _ in range(n):
    x,a=input().split()
    if a=='G':
        arr[int(x)]=0
    else:
        arr[int(x)]=1

ans=0
print(arr)
for i in range(101):
    for j in range(i,101):
        if arr[i]==-1 or arr[j]==-1:
            continue      
        cnt=0
        cur_cnt=[0,0]
        for k in range(i,j+1):
            if arr[k]!=-1:
                cur_cnt[arr[k]]+=1
        if (cur_cnt[0]==cur_cnt[1]) or not (cur_cnt[0]!=0 and cur_cnt[1]!=0):
            cnt=j-i 
        ans=max(ans,cnt)

print(ans)

File: 260110/G or H 2/G-or-H-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const arr = input.slice(1).map((a)=>a.split(' '));

const people = Array(101).fill(0);
for(let [pos, c] of arr){
    people[pos] = c;
}

let ans = 0;
for(let i=0; i<101; i++){
    if(people[i]===0) continue;
    for(let j=i; j<101; j++){
        if(people[j]===0){
            continue;
        }
        let [g, h] = [0, 0];
        for(let k=i; k<=j; k++){
            if(people[k]==='G'){
                g++;
            }else if(people[k]==='H'){
                h++;
            }
        }
        if(g===0 || h===0 || g===h){
            ans = Math.max(ans, Math.abs(j-i));
        }

    }
}

console.log(ans)